include/autoppl/expression/distribution/discrete.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include <cassert>
+#include <random>
+#include <cmath>
+#include <numeric>
+
+namespace ppl {
+
+// TODO: change name to DiscreteDist and make class template.
+// Discrete should be a function that creates this kind of object.
+
+template <typename weight_type>
+struct Discrete 
+{
+    using value_t = int;
+    using dist_value_t = double;
+
+    Discrete(std::initializer_list<weight_type> weights)
+        : weights_{weights} { assert(weights.size() > 0); }
```

Comment:
can we normalize in the constructor? Otherwise, `pdf` will recompute the normalization at every call, which is very costly.